src/lib/secret.ts

```diff
@@ -0,0 +1,51 @@
+export class Secret {
+  // A fauna key, like `fn1234`.
+  key: string;
+  // A database scope, like `["foo", "bar"]`
+  scope: string[];
+
+  constructor(opts: { key: string; scope?: string[] }) {
+    this.key = opts.key;
+    this.scope = opts.scope ?? [];
+  }
+
+  static parse(key: string) {
+    if (key.length === 0) {
+      throw new Error("Secret cannot be empty");
+    }
+    if (key.includes(":")) {
+      throw new Error("Secret cannot be scoped");
+    }
```

Comment:
I don't think so. I currently use scoped secrets by hand when I need to access a nested database, but the `fauna eval` and `fauna shell` commands both have options to set a scope and role, which should cover all of these cases.

Reply:
yes it could break existing usecases. I'll add a second parsing function for just the `--secret` argument, so you can set a scoped secret in there. I think its acceptable to disallow scoped secrets in ~/.fauna-shell though.